exp/ipam/api/v1alpha1/ipaddressclaim_types.go

```diff
@@ -0,0 +1,69 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+)
+
+// IPAddressClaimSpec is the desired state of an IPAddressClaim.
+type IPAddressClaimSpec struct {
+	// PoolRef is a reference to the pool from which an IP address should be created.
+	PoolRef corev1.TypedLocalObjectReference `json:"poolRef"`
```

Comment:
Was reviewing the enhancement and noticed this is using GroupKind to find obejcts, just wanted to note that this is [discouraged now](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#kind-vs-resource) and that the TypedLocalObjectReference exists for legacy sake. Current API conventions recommend to create your own reference types (to avoid someone adding a new API field without you noticing) and to use GroupResource for unambiguity. I the API hasn't evolved past alpha yet, it might be worth looking at making that change before it is promoted to a more stable API

Reply:
Seems to ~ fit in this discussion: https://github.com/kubernetes-sigs/cluster-api/issues/6539 ?

Reply:
Yes I think it does, we should revive that conversation, I'll see if I can re-review that thread